hardhat.config.js

```diff
@@ -11,16 +11,20 @@ for (const f of fs.readdirSync(path.join(__dirname, 'hardhat'))) {
 }
 
 const enableGasReport = !!process.env.ENABLE_GAS_REPORT;
+const enableProduction = process.env.MODE === 'PROD';
```

Comment:
```suggestion
const enableProduction = process.env.COMPILE_MODE === 'production';
```

Rationale: Using the value `'production'` which is very common with stuff like Webpack. They use the variable `NODE_ENV` but I don't think that's very accurate for this case. I changed to `COMPILE_MODE` to be a little more specific.